js/modules/resolution.go

```diff
@@ -328,3 +336,33 @@ func ExportGloballyModule(rt *sobek.Runtime, modSys *ModuleSystem, moduleName st
 		}
 	}
 }
+
+// UnknownModulesError is returned when loading a module was not possbile due to one or more of dependencies
+// that couldn't be resolved.
+type UnknownModulesError struct {
+	unknownModules []string
+}
+
+func newUnknownModulesError(list []string) UnknownModulesError {
+	slices.Sort(list)
+	return UnknownModulesError{unknownModules: list}
+}
+
+func (u UnknownModulesError) Error() string {
+	return fmt.Sprintf("unknown modules [%s] were tried to be loaded, but couldn't - "+
```

Comment:
I think there is still a bit of improvement to be done to have a better DX. A revisited version of original Ankur's proposal.

```suggestion
	return fmt.Sprintf("unknown module(s): [%s] k6 couldn't resolve the them because they are not built-in k6 modules. If these are extension modules, enable automatic extension resolution or build a custom binary with xk6. See the docs for more details: https://grafana.com/docs/k6/latest/extensions/explore/#use-extensions
```

Reply:
I think that we are going overboard on this.

There are more changes to come, and this error is either never going to be seen.  Or it will be seen before it either loads a new binary or returns a next error about extension resolutions. 

I would prefer all the DX/UX work to be done after we move to the new implementation, which already fix bugs that are reported and people have problems with. I do not think that if someone has disabled auto extension resolution, they would not know about extensions.